src/controller/index.ts

```diff
@@ -404,6 +408,67 @@ export default class SimulariumController {
         return this.playBackFile;
     }
 
+    public async getMetrics(config: NetConnectionParams): Promise<void> {
+        if (
+            !this.metricsCalculator ||
+            !this.metricsCalculator.socketIsValid()
+        ) {
+            const webSocketClient = new WebsocketClient(config, this.onError);
+            this.metricsCalculator = new RemoteMetricsCalculator(
+                webSocketClient,
+                this.onError
+            );
+            await this.metricsCalculator.connectToRemoteServer();
+        }
+        this.metricsCalculator.getAvailableMetrics();
+    }
+
+    public async getPlotData(config: NetConnectionParams): Promise<void> {
+        if (!this.simulator || this.simulator instanceof ClientSimulator) {
+            // getting plot data requires RemoteSimulator or LocalFileSimulator
+            return;
+        }
+
+        if (
+            !this.metricsCalculator ||
+            !this.metricsCalculator.socketIsValid()
+        ) {
+            const webSocketClient =
+                this.remoteWebsocketClient &&
+                this.remoteWebsocketClient.socketIsValid()
+                    ? this.remoteWebsocketClient
+                    : new WebsocketClient(config, this.onError);
+            this.metricsCalculator = new RemoteMetricsCalculator(
+                webSocketClient,
+                this.onError
+            );
+        }
+
+        // TODO: add form for users to select the metrics of hardcoding them
+        const plot1 = {
+            plotType: "scatter",
+            metricsIdx: 0,
+            metricsIdy: 2,
+            scatterPlotMode: "lines",
+        };
+        const plot2 = {
+            plotType: "histogram",
+            metricsIdx: 3,
+        };
```

Comment:
can you pass these objects in from the outside? For this they should be defined in the examples/Viewer.tsx and passed down.  (like, create some simple type for a plotconfig and pass an array of them in.)